tests/virt_autotest/prepare_non_transactional_server.pm

```diff
@@ -0,0 +1,100 @@
+# PREPARE NON-TRANSACTIONAL SERVER MODULE
+#
+# Copyright 2025 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+#
+# Summary: This module do preparation work for virtual machine installation on non
+# -transactional server, including extensions registration, packages installation,
+# services toggle and other operations being pertinent to non-transactonal server
+# or which need to to performed in advance for various purposes, including better
+# test flow, clear functional division, placeholder for future extension and etc.
+#
+# Maintainer: Wayne Chen <wchen@suse.com> qe-virt@suse.de
+package prepare_non_transactional_server;
+
+use base "opensusebasetest";
+use strict;
+use warnings;
+use testapi;
+use transactional;
+use utils;
+use version_utils;
+use Utils::Systemd;
+use Utils::Backends qw(get_serial_console);
+use ipmi_backend_utils;
+use virt_autotest::utils;
+
+sub run {
+    my $self = shift;
+
+    $self->prepare_ground;
+    $self->prepare_extensions;
+    $self->prepare_packages;
+    $self->prepare_bootloader;
+    $self->prepare_services;
+    $self->prepare_reboot;
+    $self->restore_ground;
```

Comment:
@Julie-CAO would you please give it a review if these steps have any redundancy with host installation setup from the regard of OS preparation?

Reply:
I do not think there is any duplicate.

Reply:
I found some steps are duplicate when I run test, such as generating or backing up ssh keys, restarting libvirt services ... I can't list all, I'll give more input when I have a good check.

Reply:
I meant in sle15sp7 UEFI guest tests. I didn't look at SLM tests.

Reply:
I looked at the verification run ---- http://10.200.140.9/tests/210,  there are no duplicate steps.

About `INSTALL_OTHER_REPOS`, I saw the repos from Tumbleweed and SLE15-SP5 were added, is it safe? I mean sle15sp5 has gone to LTSS, are these repos or packages are always available?

Reply:
1. Tumbleweed repo is just being used temporarily, after bsc#1236313 is fixed, I will not use it.
2. What I installed just standalone packages, so safe. Actually there is no much difference using 15-SP5 or 15-SP6 repository (because they are standalone packages), but I can try to have a look whether I can use more recent repositories. But if I can not find them in more recent repositories., I will continue to use old ones.

Reply:
@Julie-CAO  Hi Julie, I meant  if these steps have any redundancy with your sle16 host installation setup in jsonnet from the regard of OS preparation,eg usr setup, selinux, grub2,etc?

Reply:
Good point. Actually it would be better to set up these steps when installing a host, including installing packages, configuring grub2, restarting services and etc, but I have not implemented them yet. I have some concerns that the current .json file has limited function, I am not certain it can support all these requirements finally. Because I find that the current .json is not so powerful as autoyast template, it doesn't support fully passing Openqa settings, or some conditional statements.
We have to talk if these steps can be moved from here to host installation one by one.

Reply:
You can move it to host installation later one by one if you think it is necessary. But bear it in mind, this module ```tests/virt_autotest/prepare_non_transactional_server.pm``` should not be removed, because it provides customization when necessary. 
By the way, according to what I have seen, there is no ```packages``` (just ```patterns```) in ```software``` section in agama jsonnet schema **at the moment**.

Reply:
Glad to know your opinions. Let me also share mine. I think some basic guide  can be like this
- the *necessary* setups before initial connection and user login in login_console, eg serial console, sshd setup, user/passwd, firewall, etc, are put in jsonnet, and do NOT duplicate in prepare_non_transactional_server.pm
- the optional setups before login_console, is fine to be either in jsonnet or  prepare_non_transactional_server.pm, but still no duplication and will be good to be easy to debug directly in openqa

HDYT?

Reply:
It is OK if you want to keep one thing only in one place, but do not over-engineering, because what we do is for the purpose of automated test run, not for sake of doing it.